src/gui/editors/PianoRoll.cpp

```diff
@@ -961,6 +961,8 @@ void PianoRoll::shiftSemiTone( int amount ) // shift notes by amount semitones
 		}
 	}
 
+	m_pattern->dataChanged();
```

Comment:
You need to call ```m_pattern->rearrangeAllNotes();``` before `dataChanged()` as you may have moved the note past another note so they need to be sorted again.

Reply:
If I understand it correctly, `m_pattern->rearrangeAllNotes()` is for sorting notes by start time, but the main function `shiftSemiTone` doesn't affect the notes' start time (horizontal position), it only affects the notes' keys (vertical position), so `m_pattern->rearrangeAllNotes()` isn't needed, right?

Reply:
Wrong. I changed this recently because the notes need to be sorted 'vertically' in order for the arpeggiator to work. Otherwise the order of execution will be randomly. Fixed here: https://github.com/LMMS/lmms/pull/3881

Reply:
Done in https://github.com/LMMS/lmms/pull/3961/commits/31d32552964f80f0c303a47059f245d397cff287.